mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -467,11 +491,11 @@ impl Nexus {
         NexusChannel::reconfigure(self.as_ptr(), &event);
 
         let result = r.await;
-
         info!(
             "{}: Dynamic reconfiguration event: {:?} completed {:?}",
             self.name, event, result
         );
+        self.release_reconfigure();
```

Comment:
Is there a race condition here as we clear the dr_complete_notify before releasing the reconfigure. Can another reconfigure get in between? Then it'd get stuck?

Reply:
No, because if dr_complete_notify isn't set, then it won't wait on the channel either.
Also, if there is no await in between those points, I don't think another reconfigure can happen.

Reply:
Yeah true but what if more than 2 extra reconfigure come along? the first goes in waits on r.await (493); then the callback "takes" the dr_complete_notify and does s.send, but before the first r.await another 2 reconfigure come along, the first one goes through and the second awaits on the oneshot (465)... meanwhile the initial reconfigure gets to run again and releases the oneshot (465) channel and now we have 2 reconfigure running?

Reply:
I may be misunderstanding this but then again concurrency is hard :) I had a play with the async mutex and it "seems" to work, so maybe worth giving investing in that one a bit more?